### ISSUE TITLE
Add a normalization component

### ISSUE THREAD

**Issue by eccabay** [alteryx/evalml#3355]
Part of the work for supporting unsupervised learning in evalml, we need a normalization component to ensure features on different scales don't negatively affect the performance of models.

**Comment by eccabay** [comment]
[Scikit-learn's MinMaxScaler](https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.MinMaxScaler.html) looks like it does exactly what we're looking for here
